addons/website_sale_loyalty/controllers/main.py

```diff
@@ -130,3 +130,21 @@ def cart_update_json(self, *args, set_qty=None, **kwargs):
             # and does not follow the request's context
             request.website = request.website.with_context(website_sale_loyalty_delete=True)
         return super().cart_update_json(*args, set_qty=set_qty, **kwargs)
+
+
+class PaymentPortal(main.PaymentPortal):
+
+    def _validate_transaction_for_order(self, transaction, sale_order_id):
+        """Update programs & rewards before finalizing transaction.
+
+        :param payment.transaction transaction: The payment transaction
+        :param int order_id: The sales order to pay, as a `sale.order` id
+        :raise: ValidationError the order amount changed after updating rewards
+        """
+        super()._validate_transaction_for_order(transaction, sale_order_id)
+        order = request.website.sale_get_order()
```

Comment:
```suggestion
        order = request.env['sale.order'].browse(sale_order_id)
```
Let's be a bit more straightforward ^^